pkg/apis/core/v1alpha1/extensions.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ExtensionRequestSpec specifies details about a request to add an extension to Crossplane.
+type ExtensionRequestSpec struct {
+	Source                   string `json:"source,omitempty"`
+	Package                  string `json:"package,omitempty"`
+	CustomResourceDefinition string `json:"crd,omitempty"`
+}
```

Comment:
I'm reading this here first before looking at the doc. Comments for these `string` fields might make it more obvious what they were for.